MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs

```diff
@@ -729,7 +727,7 @@ public async Task<string> GetSubtitleFileCharacterSet(string path, string langua
         {
             var bytes = await GetBytes(path, protocol, cancellationToken).ConfigureAwait(false);
 
-            var charset = _textEncoding.GetDetectedEncodingName(bytes, bytes.Length, language, true);
+            var charset = CharsetDetector.DetectFromBytes(bytes).Detected?.EncodingName;
```

Comment:
What this does? Seems to be repeating some initialization upon subsequent calls, is it what you wanted?

Reply:
What do you mean? EncodingName should be the name of the code page, which it uses as a parameter to ffmpeg ` encodingParam = " -sub_charenc " + encodingParam;`

Reply:
Here's an example: https://dotnetfiddle.net/BmBd2B